📊 docs: helm chart 2.x initial (#231)
LibreChat-AI/librechat.ai: pages/docs/local/helm_chart.mdx

# Deployment a LibreChat Helm Chart

The following instructions guide you to deploy LibreChat on Kubernetes using Helm.

Note: this method was contributed by the community. If you are familiar with Helm and Kubernetes, use this guide as a reference and adjust to your needs. You can also reference other helm charts made by the community below in the [Community Helm Charts](#community-helm-charts) section.

## Prerequisites

* A running Kubernetes cluster
* `kubectl` installed
* Helm installed on your local machine

## Configuration

> 1. Generate Variables Generate CREDS_KEY, JWT_SECRET, JWT_REFRESH_SECRET and MEILI_MASTER_KEY using openssl rand -hex 32 and CREDS_IV using openssl rand -hex 16. place them in a secret like this (If you want to change the secret name, remember to change it in your helm values):

Reviewer: ```suggestion
1. Use the [Credentials Generator](/toolkit/creds_generator) to quickly generate secure values for `CREDS_KEY`, `JWT_SECRET`, `JWT_REFRESH_SECRET` and `MEILI_MASTER_KEY`.
Place them in a Kubernetes Secret like this (If you want to change the secret name, remember to change it in your helm values):
```

```yaml
apiVersion: v1
kind: Secret
metadata:
  name: librechat-credentials-env
  namespace: <librechat-chart-namespace>
type: Opaque
stringData:
  CREDS_KEY: <generated value>
  JWT_SECRET: <generated value>
  JWT_REFRESH_SECRET: <generated value>
  MEILI_MASTER_KEY: <generated value>
```

> 2. Add Credentials to the Secret Dependant of the Model you want to use, create Credentials in your provider and add them to the Secret:

Reviewer: ```suggestion
2. Depending on the provider you wanna use, add the respective credentials to the Kubernetes secret as well:
```
